### PR TITLE
Reconcile all in-cluster resources via UCCM

### LINKED ISSUE #3122: Move reconciling of all in-cluster resources from the cluster controller into the user cluster controller
**User Story**

We still create some objects inside the user cluster from within the cluster controller. This has the drawback, that we don't get notified when something changes, leading to potential downtimes. 

**Acceptance criteria**

* The cluster-controller does not create anything inside the user cluster

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Moves reconciling of the remaining resource (openvpn client cert) into the UCCM, making the cloud controller not do anything inside the user cluster controller.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3122 

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
none
```

### CODE REVIEW

File: api/pkg/controller/usercluster/controller.go [hunk 1]
```diff
@@ -38,16 +40,17 @@ func Add(
...
 	clusterURL *url.URL,
 	openvpnServerPort int,
-	registerReconciledCheck func(name string, check healthcheck.Check)) error {
+	registerReconciledCheck func(name string, check healthcheck.Check),
+	openVPNCA *resources.ECDSAKeyPair) error {
 	reconciler := &reconciler{
 		Client:            mgr.GetClient(),
 		cache:             mgr.GetCache(),
 		openshift:         openshift,
-		rLock:             &sync.Mutex{},
```
Comment (alvaroaleman) [inline api/pkg/controller/usercluster/controller.go]: I believe this is not needed anymore as we use the same workqeue key for all types

Comment (mrIncompetent) [reply api/pkg/controller/usercluster/controller.go]: The mutex has nothing todo with the workqueue.
It protects `r,reconciledSuccessfullyOnce` from being accessed concurrently.

**Comment by alvaroaleman** [comment]
@mrIncompetent Updated, PTAL

**Pull request merged**